components/chainhook-cli/src/scan/bitcoin.rs

```diff
@@ -98,9 +98,14 @@ pub async fn scan_bitcoin_chainstate_via_rpc_using_predicate(
     let mut last_scanned_block_confirmations = 0;
     let http_client = build_http_client();
 
+    let mut loop_did_trigger = false;
     while let Some(current_block_height) = block_heights_to_scan.pop_front() {
         if let Some(ref mut predicates_db_conn) = predicates_db_conn {
-            if number_of_blocks_scanned % 10 == 0 || number_of_blocks_scanned == 0 {
+            if number_of_blocks_scanned % 10 == 0 
+                || number_of_blocks_scanned == 0
+                // if the last loop did trigger a predicate, update the status
+                || loop_did_trigger
```

Comment:
Can't this conditional be replaced by `number_of_times_triggered > 1`?

Reply:
No, they are not interchangeable. `number_of_times_triggered ` is a running total of the number of triggers. `loop_did_trigger` identifies if the _last loop_ triggered.

The goal of this PR is that the status that is stored in redis always stays updated with the latest triggers. This will prevent refiring a trigger on restart. (side note, we can probably now increase that mod to `number_of_blocks_scanned % 100`. We don't need to update redis as often for non-triggered predicates if we're sure we always keep updated every time we trigger.

If we used `number_of_times_triggered > 1`, we'd always update the status after a single trigger has happened for a predicate, which would be a lot of calls to redis.

Does that makes sense? We can get on a call if this isn't clear

Reply:
Alright, thanks for clearing that up, Micaiah.